lib/java/src/main/java/org/apache/thrift/server/TSaslNonblockingServer.java

```diff
@@ -171,6 +171,7 @@ private void acceptNewConnection() {
             LOGGER.warn("Failed to accept incoming connection.", e);
           }
         } else {
+          // TODO fix with try-with-resource for the return value of .channel
```

Comment:
TODOs in code are easily forgotten. Better to create an issue in the issue tracker for the project than to embed a TODO.

Reply:
let me revert this line - I don't think I can fully test whether the selector resource _will_ leak. let me re-check later.
```suggestion
```